compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/tower/FirTowerResolverSession.kt

```diff
@@ -564,10 +563,15 @@ class FirTowerResolverSession internal constructor(
 
         for (invokeReceiverCandidate in candidateFactoriesAndCollectors.invokeReceiverCollector!!.bestCandidates()) {
             val symbol = invokeReceiverCandidate.symbol
-            if (symbol !is FirCallableSymbol<*> && symbol !is FirRegularClassSymbol) continue
+            if (symbol !is FirCallableSymbol<*> && symbol !is FirRegularClassSymbol && symbol !is FirTypeAliasSymbol) continue
 
-            val isExtensionFunctionType =
-                (symbol as? FirCallableSymbol<*>)?.fir?.returnTypeRef?.isExtensionFunctionType(components.session) == true
+            val returnTypeRef =
```

Comment:
Suggest rewriting as:
```kt
val returnTypeRef = when (symbol) {
    is FirTypeAliasSymbol -> symbol.fir.expandedTypeRef
    is FirCallableSymbol<*> -> symbol.fir.returnTypeRef
    is FirRegularClassSymbol -> null
    else -> continue
}
```

Reply:
Even `symbol` can be moved into `when`. Much simpler now, thanks!